api/handlers_instances.go

```diff
@@ -659,7 +668,7 @@ func (s *server) InstanceProfileDeleteHandler(w http.ResponseWriter, r *http.Req
 		role:         fmt.Sprintf("arn:aws:iam::%s:role/%s", account, s.session.RoleName),
 		inlinePolicy: policy,
 		policyArns: []string{
-			"arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess",
+			"arn:aws:iam::aws:policy/IAMReadOnlyAccess",
```

Comment:
Do we even need this? You are already enumerating the specific Get/List policy permissions above that should be sufficient. I think we just leave `policyArns` empty - the less permissions the better, especially when using IAM